### ISSUE TITLE
Learning Rate Issue

### ISSUE THREAD

**Issue by menelaus02** [adriangb/scikeras#270]
I am trying to adjust my lr to find the best parameter. When I  use tf.keras.wrappers.scikit_learn, it can work. But  when I use scikeras, I can't adjust lr.
I use 
`model = KerasRegressor(model=network_model)
GridSearchCV(estimator=model,param_grid=param_grid,scoring='neg_mean_squared_error',cv=5)`

**Comment by adriangb** [comment]
Would you mind giving a minimal reproducible example?

**Comment by menelaus02** [comment]
`def network_model(num=256,learning_rate=0.01,IMSIZE=128):
    model=Sequential() 
    model.add(Flatten(input_shape=(IMSIZE,IMSIZE,3))) 
    model.add(Dense(num,activation='relu')) 
    model.add(Dense(1)) 
    model.compile(loss='mse',optimizer=Adam(learning_rate),metrics=['mse']) 
    model.summary()
    return model  
from scikeras.wrappers import KerasRegressor
from sklearn.model_selection import GridSearchCV

param_grid = {
        "num":[64,128,256,512,1024],
        "learning_rate":[0.001,0.005,0.01,0.05],
        'batch_size':[128,256,512],
        'epochs':[30,50,100]
        }

model = KerasRegressor(model=network_model) 

grid_search = GridSearchCV(estimator=model,param_grid=param_grid,scoring='neg_mean_squared_error',cv=5) 
grid_search.fit(x_train,y_train) 
`

**Comment by adriangb** [comment]
Thanks. I'm on vacation for the next couple days, I'll get around to this as soon as I can.

**Comment by adriangb** [comment]
Please take a look at this notebook: https://colab.research.google.com/drive/1F612ADj4QDA1S4hMzWnNQB0yI8evl21D#scrollTo=tQP7rWirvaUp

I took your example and simplified it a bit. I see that the learning rate _is_ being changed.

**Comment by adriangb** [comment]
Hi @menelaus02, looping back here. Were you able to resolve your issue?

**Comment by menelaus02** [comment]
> Hi @menelaus02, looping back here. Were you able to resolve your issue?

Hi, I have solved it. I found that I ought to add the initial value of parameters to the KerasRegressor so that it could run normally.  Otherwise it woud apper "Check the list of available parameters with `estimator.get_params().keys()`”.
`
model = KerasRegressor(model=network_model, verbose=False) 
model = KerasRegressor(model=network_model, learning_rate=0.1, verbose=False) 
`

**Comment by adriangb** [comment]
Excellent, I'm glad you could get it to work. Are there any extra warnings or documentation you think might have helped catch this earlier?

**Comment by menelaus02** [comment]
Maybe we can mention in the documentation that if we compile the model and use the optimizer in a function, we need to initialize the value of the parameters even if we use the default parameters in the function.

**Comment by adriangb** [comment]
Does you feel that this section cover the issue? https://www.adriangb.com/scikeras/stable/migration.html#id8

**Comment by menelaus02** [comment]
I think it is clear enough.

**Comment by adriangb** [comment]
Great, would you mind closing the issue then?

**Comment by menelaus02** [comment]
No problem.
